[ADD] accounting: add new inventory valuation page with 19.0 updates (#17011)
odoo/documentation: content/applications/finance/accounting/get_started/inventory-valuation.rst

===================
Inventory valuation
===================

A company’s inventory valuation should include all stock on hand. Accurately recording this value
in the accounting records ensures a true representation of the company's current asset value.

The :doc:`Inventory <../../../inventory_and_mrp/inventory>` app maintains real-time stock valuation
based on physical item movement. In contrast, the Accounting app updates accounting inventory
valuation only when specifically requested, most often for closing or in perpetual accounting, when
vendor bills or customer invoices are posted.

.. seealso::
   Use the :doc:`../../../inventory_and_mrp/inventory/inventory_valuation/cheat_sheet` for guidance.

.. _accounting/inventory-valuation/accounting-standards:

Accounting standards
====================

In accounting, the Continental and Anglo-Saxon methods differ in when they recognize inventory

> expenses and how this impacts :ref:`journal entries <accounting-entries>`:

Reviewer: ```suggestion
In accounting, the timing of inventory expense recognition and its impact on :ref:`journal entries <accounting-entries>` differ based on the accounting method used:
```

- Using Continental accounting records the cost of goods as an expense when the vendor bill is
  posted, typically upon receipt into stock, regardless of when they are sold.
- Using Anglo-Saxon accounting recognizes the cost of goods sold (COGS) as an expense when the

>   customer invoice is posted, typically when products are sold.

Reviewer: ```suggestion
- **Continental** accounting: the cost of goods is recorded as an expense when the vendor bill is
  posted, typically upon receipt into stock, regardless of when the goods are sold.
- **Anglo-Saxon** accounting: the cost of goods sold (COGS) is recognized as an expense when the
  customer invoice is posted, typically when products are sold.
```

  method as :guilabel:`Perpetual (at invoicing)` or :guilabel:`Periodic (at closing)`.
- :guilabel:`Periodic Valuation`: Set the :ref:`closing entry
  <accounting/inventory-valuation/closing-entry>` process to :guilabel:`Manual`, :guilabel:`Daily`,
  or :guilabel:`Monthly` to automate it, if needed.
- :ref:`Inventory Cost Method <costing-methods>`: Choose :guilabel:`Standard Price`,
  :guilabel:`First in First Out (FIFO)`, or :guilabel:`Average Cost (AVCO)`.
- :ref:`Valuation Account <accounting/inventory-valuation/valuation-account>`: Set the asset account
  used to record the financial value of physical stock. Then, click the :icon:`oi-arrow-right`
  :guilabel:`(right arrow)` in the :guilabel:`Valuation Account` field to open the :guilabel:`Stock
  Valuation` account and update the :ref:`Variation Account
  <accounting/inventory-valuation/variation-account>`.
- :guilabel:`Journal`: Set the journal to post inventory valuation entries in.
- :guilabel:`Other accounts can be defined on "Inventory Loss" and "Production" on dedicated
  locations`: If needed, set the :ref:`Inventory Loss
  <accounting/inventory-valuation/inventory-loss>` or the :ref:`Cost of Production

>   <accounting/inventory-valuation/cost-of-production>` accounts.

Reviewer: ```suggestion
  locations`: If needed, click :guilabel:`locations` and set the :ref:`Inventory Loss
  <accounting/inventory-valuation/inventory-loss>` or the :ref:`Cost of Production
  <accounting/inventory-valuation/cost-of-production>` accounts.
```

.. note::

>    Default accounts, valuation, and cost methods can be overridden for each product category.

Reviewer: do we have a link for how to do this, maybe?

Reviewer 2: So, we don't. But if we change it to "on" instead of "for", does that do enough? It's really as simple as setting those fields on the product categories, so explaining it seems redundant ("accounts etc. can be overridden for each product category by setting the accounts etc. on the product category")

Reviewer: what about adding *form* as well then?

"accounts etc. can be overridden for each product category by setting them on the product category form."

(and I feel like this should be a tip)

.. tip::
   In :guilabel:`Perpetual` valuation, it is more common to set the :guilabel:`Periodic Valuation`
   to :guilabel:`Manual` and generate closing entries only when a fiscal period closes, or when
   accounting reports require synchronization with inventory stock, unless an automatic update of

>    accounting inventory value is needed.

Reviewer: I think this tip would maybe fit better under the next section, after we've explained  the difference between the two types of inventory valuation? otherwise I feel like this lacks context a little bit

Reviewer 2: I've added it to the note at the bottom of the next section instead of the tip because I thought it fit better there. LMK if you disagree :)

.. _accounting/inventory-valuation/valuation-method:

Valuation methods
-----------------

Two accounting practices are used for maintaining inventory records, each differing in how and when

> inventory costs are recognized:

Reviewer: I'd like to rephrase the second part of this sentence and avoid the "how and when" if possible.

```suggestion
The timing and method used to maintain inventory records vary based on the accounting practice:
```
but then: 
- Is *practice* the most appropriate term here?
- I struggle with finding a solution here so if you have a better idea, feel free to use it!

- :guilabel:`Periodic (at closing)` updates inventory valuation only when generating entries during

>   the :ref:`stock closing process <accounting/inventory-valuation/closing-entry>`. While

Reviewer: ```suggestion
- :guilabel:`Periodic (at closing)` inventory valuation is updated only when generating entries during
  the :ref:`stock closing process <accounting/inventory-valuation/closing-entry>`. While
```

  inventory movements are tracked physically via the :doc:`Inventory Stock
  <../../../inventory_and_mrp/inventory/warehouses_storage/reporting/stock>` report, they are not

>   automatically synchronized with financial records. In :guilabel:`Periodic`, the Expenses account

Reviewer: the *in* sounds odd to me

```suggestion
  automatically synchronized with financial records. When using :guilabel:`Periodic` inventory valuation, the Expenses account
```

  is debited when vendor bills are posted.

> - :guilabel:`Perpetual (at invoicing)` updates inventory valuation when bills or invoices are

Reviewer: ```suggestion
- :guilabel:`Perpetual (at invoicing)` inventory valuation is updated when bills or invoices are
```

  posted. The Expenses account (:abbr:`COGS (Cost of goods sold)`) is debited when invoices are
  posted. Combined with the :ref:`stock closing process
  <accounting/inventory-valuation/closing-entry>`, it updates inventory value for deliveries and

>   receipts, even if there is no invoice or vendor bill yet.

Reviewer: I'm not 100% what *it* refers to here, do we mean the Perpetual inventory valuation? and I think i'd rephrase to maybe switch to the passive voice here, too

.. note::
   The periodic accounting method is often associated with Continental standards, while the
   perpetual method is commonly used with Anglo-Saxon standards. However, companies may choose a
   different accounting valuation method based on their specific needs.

.. tip::

>    When using :guilabel:`Perpetual (at invoicing)` valuation method with the :guilabel:`Standard

Reviewer: ```suggestion
   When using the :guilabel:`Perpetual (at invoicing)` valuation method with the :guilabel:`Standard
```

   Price` costing method, a :guilabel:`Price Difference Account` can be set in the
   :guilabel:`Product Category`. This account records the difference between the product's standard
   price, as specified on the product form, and the actual billed price, helping track pricing
   fluctuations.

.. _accounting/inventory-valuation/valuation-account:

Valuation account
-----------------

The :guilabel:`Valuation account` records the inventory value listed as a current asset on the
balance sheet. How the :guilabel:`Valuation Account` tracks inventory asset value depends on the

> selected :ref:`valuation method <accounting/inventory-valuation/valuation-method>`.

Reviewer: ```suggestion
balance sheet. The :guilabel:`Valuation Account` tracks inventory asset value differently depending on the selected
:ref:`valuation method <accounting/inventory-valuation/valuation-method>`.
```

.. _accounting/inventory-valuation/variation-account:

Variation account
~~~~~~~~~~~~~~~~~

The :guilabel:`Variation account` is used to record inventory variations for the period covered by
the :ref:`stock closing process <accounting/inventory-valuation/closing-entry>` and can be updated
during :ref:`configuration <accounting/inventory-valuation/configuration>`:

- Continental :guilabel:`Periodic (at closing)`: Expense account (specific for variation recording).
- Anglo-Saxon :guilabel:`Perpetual (at invoicing)`: Current asset account (recommended for interim
  tracking) or expense account (optional).

  .. note::
     - Stock variations resulting from product receipts and deliveries are displayed in the
       :guilabel:`Stock Variation` section of the :ref:`Inventory Valuation report
       <accounting/inventory-valuation/inventory-valuation-report>`.

>      - In perpetual accounting, if no bill or invoice has been posted for these moves yet, it allows

Reviewer: probably a very dumb question, sorry! but perpetual accounting is the same thing as perpetual inventory valuation (and all the other terms used throughout the doc, see [this comment ](https://github.com/odoo/documentation/pull/17011/changes#r3585451826)), correct?

Reviewer 2: very reasonable question - yes it's the same and I've updated it throughout.

       their impact on inventory value to be recorded in the accounting. The :guilabel:`Variation
       account` serves as a buffer account until the :ref:`accrual entries
       <accounting/inventory-valuation/accrual-entries>` are recorded.

.. _accounting/inventory-valuation/other-accounts:

Other accounts
--------------

.. _accounting/inventory-valuation/inventory-loss:

Inventory Loss
~~~~~~~~~~~~~~

To exclude :doc:`inventory adjustments
<../../../inventory_and_mrp/inventory/warehouses_storage/inventory_management/count_products>` from
the :guilabel:`Variation account` at closing, configure an :guilabel:`Inventory Adjustment` location
account (:guilabel:`Inventory Loss` location type):

#. In the :guilabel:`Other accounts can be defined on "Inventory Loss" and "Production" on dedicated
   locations` :ref:`configuration <accounting/inventory-valuation/configuration>`, click

>    :guilabel:`locations`.

Reviewer: ```suggestion
#. In the :ref:`Inventory Valuation section of the Accounting settings <accounting/inventory-valuation/configuration>`, click :guilabel:`locations` in :guilabel:`Other accounts can be defined on "Inventory Loss" and "Production" on dedicated locations`.
   
```

#. In the :guilabel:`Locations` list view, open any locations with the :guilabel:`Inventory Loss`
   location type.
#. Set the :guilabel:`Loss Account` in the :guilabel:`Accounting` information section.

.. note::
   - In Continental :guilabel:`Periodic (at closing)`, the update is displayed in the
     :guilabel:`Inventory Loss` section of the :ref:`Inventory Valuation report
     <accounting/inventory-valuation/inventory-valuation-report>`.

>    - In Anglo-Saxon :guilabel:`Perpetual (at invoicing)`, the :ref:`valuation account

Reviewer: I'd rephrase to add the appropriate term after Periodic and Perpetual, it sounds a bit odd just as is IMO

Reviewer 2: LMK what you think of my update :)

     <accounting/inventory-valuation/valuation-account>` updates immediately in real time once the
     move is completed.

.. _accounting/inventory-valuation/cost-of-production:

Cost of Production
~~~~~~~~~~~~~~~~~~

To record production costs in a different account than the general set :guilabel:`Variation
account`, for both the finished product and the components, set up a :guilabel:`Production` location

> account (:guilabel:`Production` location type):

Reviewer: ```suggestion
To record production costs for both the finished product and the components in an account other than the default :guilabel:`Variation Account`, set up a :guilabel:`Production` location account (:guilabel:`Production` location type):
```

#. In the :guilabel:`Other accounts can be defined on "Inventory Loss" and "Production" on dedicated
   locations` :ref:`configuration <accounting/inventory-valuation/configuration>`, click

>    :guilabel:`locations`.

Reviewer: ```suggestion
#. In the :ref:`Inventory Valuation section of the Accounting  settings <accounting/inventory-valuation/configuration>`, click :guilabel:`locations` in :guilabel:`Other accounts can be defined on "Inventory Loss" and "Production" on dedicated locations`.
```

#. In the :guilabel:`Locations` list view, open any locations with the :guilabel:`Production`
   location types.
#. Set the :guilabel:`Cost of Production` in the :guilabel:`Accounting` information section.

.. note::
   - In Anglo-Saxon :guilabel:`Perpetual (at invoicing)`, the :guilabel:`Valuation account` for
     finished products and components are updated when a manufacturing order is validated, and the
     :guilabel:`Cost of Production` account serves as the counterpart account for both.

>    - In Continental :guilabel:`Periodic (at closing)`, inventory variations from production (for

Reviewer: [same as above](https://github.com/odoo/documentation/pull/17011/changes#r3585511703)

Reviewer 2: lmk what you think of my changes

     both finished products and components) are recorded separately from general inventory
     variations recorded at closing, isolating production-related changes from receipts and
     deliveries. However, this account is not commonly used in standard Continental
     :guilabel:`Periodic (at closing)` accounting.

.. _accounting/inventory-valuation/inventory-valuation-report:

Inventory valuation report
==========================

The :guilabel:`Inventory Valuation` report provides an accurate valuation of inventory. To access
it, go to :menuselection:`Accounting --> Review --> Inventory Valuation`. The report includes
the following sections:

- :guilabel:`Initial Balance`: Current accounting inventory value. Click to display the stock

>   valuation journal entries.

Reviewer: Click what here exactly?

- :guilabel:`Inventory Loss` (:guilabel:`Periodic (at closing)` only): Only if the :ref:`Loss
  Account <accounting/inventory-valuation/inventory-loss>` was filled in during configuration.
- :guilabel:`Cost of Production` (:guilabel:`Periodic (at closing)` only, even if using this account
  is not common for this accounting method): Only if the :ref:`Cost of Production
  <accounting/inventory-valuation/cost-of-production>` account was entered during configuration.
- :guilabel:`Stock Variation` displays the difference between the posted inventory value and the
  stock valuation recorded in the Inventory app (i.e., the remaining quantity in stock moves).
- :guilabel:`Ending Stock`: Future accounting inventory value when the :ref:`closing entry

>   <accounting/inventory-valuation/closing-entry>` is completed. Click to access the :doc:`Inventory

Reviewer: i'd specify what we need to click exactly

   :icon:`fa-calendar` :guilabel:`As of` to select a different inventory valuation date.
#. Click :guilabel:`Generate Entry`.
#. Review the draft :guilabel:`Stock Closing` entry if needed, and click :guilabel:`Post`.

.. note::
   The :guilabel:`Stock Valuation` and :guilabel:`Stock Variation` accounts are then updated in the
   :ref:`general ledger <accounting/reporting/general-ledger>`.

.. _accounting/inventory-valuation/accrual-entries:

Accrual entries
===============

Revenue and expenses are recognized when goods or services are delivered or received, not when
invoiced. Accrual entries are required when the delivery timing differs from invoicing, especially
when a closing is needed.

To check for pending transactions requiring accrual entries:

#. Go to :menuselection:`Accounting --> Review` and select the relevant report: :guilabel:`Bill To
   Receive`, :guilabel:`Invoices To Be Issued`, :guilabel:`Billed Not Received`, and
   :guilabel:`Invoiced Not Delivered`.
#. Click :icon:`fa-calendar` :guilabel:`As of` to change the date, if needed.
#. Select the relevant lines and click :guilabel:`Create Accrual Entries`.

> #. In the :guilabel:`Accrued Revenue/Expense Entry` window, set the :guilabel:`Accrual Account` and

Reviewer: ```suggestion
#. In the :guilabel:`Accrued Revenue/Expense Entry` popup, set the :guilabel:`Accrual Account` and
```

   review the :guilabel:`Reversal Date`, if needed. Then, click :guilabel:`Create Entry`.

.. note::
   Both valuation methods use accrual entries to update revenues and expenses. However, in

>    Anglo-Saxon :guilabel:`Perpetual (at invoicing)`, these entries also take value from the buffer

Reviewer: same as [this comment](https://github.com/odoo/documentation/pull/17011/changes#r3585511703)

=========================================

In Odoo 19, the stock input/output accounts are no longer used. Any non-zero balances in stock
interim accounts must be transferred to the :ref:`stock valuation account
<accounting/inventory-valuation/valuation-account>` via journal entry to maintain accurate inventory
values after migration. This rebalancing can be performed :ref:`before
<accounting/inventory-valuation/upgrade-process-server-action>` or :ref:`after
<accounting/inventory-valuation/upgrade-process-manual>` upgrading to Odoo 19.

Before upgrading, review received purchase or sales orders that are not yet linked to vendor bills
or invoices, and post any existing draft bills/invoices to reduce the open balance.

.. tip::
   It is recommended to implement the change **after** upgrading to Odoo 19, because a server action
   identifies the open balance in the :guilabel:`Stock Interim` accounts and automatically applies
   the change.

.. important::
   Account structures and valuation scenarios differ across companies. It is recommended to review
   these steps with an accountant or someone experienced with Odoo and inventory valuation to ensure

>    they align with the company's specific setup.

Reviewer: ```suggestion
   Account structures and valuation scenarios differ across companies. Review these steps with an accountant or an Odoo inventory valuation expert to ensure the setup aligns with
the company's specific needs.
```

.. _accounting/inventory-valuation/upgrade-process-server-action:

Option 1: Server action journal entry
-------------------------------------

.. important::

>    It’s recommended first to use a testing database created during the upgrade process and verify

Reviewer: ```suggestion
   Use a test database created during the upgrade process first, and verify
```

   the :doc:`Stock report
   <../../../inventory_and_mrp/inventory/warehouses_storage/reporting/stock>`, accounts, and
   :ref:`Inventory Valuation report <accounting/inventory-valuation/inventory-valuation-report>` to
   ensure accuracy.

> After upgrading to Odoo 19, the server action must be applied per company to balance any non-zero

Reviewer: ```suggestion
After upgrading to Odoo 19, the server action must be applied for each company company to balance any non-zero
```
or to? not sure

:guilabel:`Stock Interim` accounts:

> #. Go to :menuselection:`Settings --> Users & Companies --> Companies` and access the company form.

Reviewer: ```suggestion
#. Go to :menuselection:`Settings --> Users & Companies --> Companies` and click the relevant company.
```

#. Click the :icon:`fa-cog` :guilabel:`(gear)` icon and select :guilabel:`Stock Valuation rebalance
   interim Accounts`. A draft journal entry is automatically generated to balance the open amount.

> #. Review the draft journal entry amounts and edit them if necessary, then click :guilabel:`Post`.

Reviewer: ```suggestion
#. Review the draft journal entry amounts, edit them if necessary, then click :guilabel:`Post`.
```

#. Check the General Ledger to confirm that the :guilabel:`Stock Interim` accounts are balanced to
   zero and that the remaining amount is recorded in the :guilabel:`Stock valuation` account.
#. Go to :menuselection:`Accounting --> Review --> Inventory Valuation` and :ref:`generate an
   inventory variation entry <accounting/inventory-valuation/closing-entry>`. The increased value on
   the :ref:`Valuation account <accounting/inventory-valuation/valuation-account>` is then
   transferred to the :ref:`Variation account <accounting/inventory-valuation/variation-account>`,
   which is a new type of interim account.

> #. (optional) Generate :ref:`accrual entries <accounting/inventory-valuation/accrual-entries>` if an

Reviewer: ```suggestion
#. Optionally, generate :ref:`accrual entries <accounting/inventory-valuation/accrual-entries>` if an
```

>    accounting closing is required and the Bill to Receive or GRNI account, etc., need updating.

Reviewer: ```suggestion
   accounting closing is required and the Bill to Receive or GRNI account, etc., needs updating.
```

.. tip::
   Journal entries can also be created manually if needed, using the same process for
   balancing :guilabel:`Stock Interim` accounts :ref:`before upgrading to Odoo 19
   <accounting/inventory-valuation/upgrade-process-manual>`.

.. _accounting/inventory-valuation/upgrade-process-manual:

Option 2: Manual journal entry
------------------------------

To create the manual journal entry, follow these steps:

#. If all vendor bills or invoices can’t be created for all the received purchase/sales orders
   before the upgrade, follow these steps:

   - Go to :menuselection:`Accounting --> Reporting --> General Ledger`.
   - Identify the open balance in the :guilabel:`Stock Interim` accounts and create a journal
     entry that debits/credits:

     - The :guilabel:`Stock Interim` account(s) containing the open balance
     - The :guilabel:`Stock Valuation` account

   - Check the General Ledger to confirm that the :guilabel:`Stock Interim` accounts are balanced to

>      zero and that the remaining amount is recorded in the :guilabel:`Stock valuation` account.

Reviewer: I'd use a nested numbered list here instead, where appropriate:

```suggestion
   a. Go to :menuselection:`Accounting --> Reporting --> General Ledger`.
   #. Identify the open balance in the :guilabel:`Stock Interim` accounts and create a journal
     entry that debits/credits:

     - The :guilabel:`Stock Interim` account(s) containing the open balance
     - The :guilabel:`Stock Valuation` account

   #. Check the General Ledger to confirm that the :guilabel:`Stock Interim` accounts are balanced to
     zero and that the remaining amount is recorded in the :guilabel:`Stock valuation` account.
```

#. After upgrading to Odoo 19, go to :menuselection:`Accounting --> Review --> Inventory Valuation`
   and :ref:`generate an inventory variation entry <accounting/inventory-valuation/closing-entry>`.
   The increased value on the :ref:`Valuation account
   <accounting/inventory-valuation/valuation-account>` is then transferred to the :ref:`Variation
   account <accounting/inventory-valuation/variation-account>`, which is a new type of interim
   account.

> #. (optional) Generate :ref:`accrual entries <accounting/inventory-valuation/accrual-entries>`

Reviewer: ```suggestion
#. Optionally, generate :ref:`accrual entries <accounting/inventory-valuation/accrual-entries>`
```
